jaxley/modules/cell.py

```diff
@@ -203,17 +210,26 @@ def _init_morph_jax_spsolve(self):
         """
 
         # Edges between compartments within the branches.
-        # `[offset, offset, 0]` because we want to offset `source` and `sink`, but
-        # not `type`.
         self._comp_edges = pd.concat(
             [
-                [offset, offset, 0] + branch._comp_edges
-                for offset, branch in zip(self.cumsum_nseg, self.branch_list)
+                pd.DataFrame().from_dict(
+                    {
+                        "source": list(range(cumsum_nseg, nseg - 1 + cumsum_nseg))
+                        + list(range(1 + cumsum_nseg, nseg + cumsum_nseg)),
+                        "sink": list(range(1 + cumsum_nseg, nseg + cumsum_nseg))
+                        + list(range(cumsum_nseg, nseg - 1 + cumsum_nseg)),
+                    }
+                )
+                for nseg, cumsum_nseg in zip(self.nseg_per_branch, self.cumsum_nseg)
             ]
         )
-        # `branch_list` is not needed anymore because all information it contained is
-        # now also in `self._comp_edges`.
-        del self.branch_list
+        self._comp_edges["type"] = 0
+
+        # Casting source and sink to int. This is necessary for branches which have
+        # just a single compartment, because in these cases, `_comp_edges` is empty
+        # and is instantiated as `float` by default.
+        self._comp_edges["source"] = self._comp_edges["source"].astype(int)
+        self._comp_edges["sink"] = self._comp_edges["sink"].astype(int)
```

Comment:
unless you instantiate the `_comp_edges` dataframe with `dtype=int` from the get go no?